### PR TITLE
Code gen

### PR DESCRIPTION
**Pull request by pranavkm**
Follow up from the discussion in https://github.com/dotnet/aspnetcore/pull/40342

### CODE REVIEW

**Review by dougbu — APPROVED** [review]
This looks fine to me (as long as it works 😺).

Curious: How did you decide between the separate TrimmableProject.props option and adding `%(TrimmableProject)` metadata in ProjectReferences.props❔

**Review by JamesNK — APPROVED** [review]
Love it.

Out of curiosity, do you have an aspiration to eliminate the component's WASM linking test project? Setting the WASM build properties is simple enough. The more difficult part is distinguishing between trimmable WASM projects and trimmable aspnetcore projects.

File: eng/CodeGen.proj [hunk 1]
```diff
@@ -73,6 +74,29 @@
...
+    <PropertyGroup>
+      <TrimmableProjectsList>$(MSBuildThisFileDirectory)TrimmableProjects.props</TrimmableProjectsList>
+      <TrimmableProjectsListContent>
+      <![CDATA[
+<!--
+  This file is automatically generated. Run `./eng/scripts/GenerateProjectList.ps1` to update.
+
+  This file contains a complete list of projects annotated for trimming.
+
+  This project is generated using the <Trimmable> properties from each .csproj in this repository.
```
Comment (dougbu) [inline eng/CodeGen.proj:87]: nit (just if validation build fails): a `*.props` file isn't a project
```suggestion
  This file is generated using the <Trimmable> properties from each .csproj in this repository.
```

**Comment by pranavkm** [comment]
> Curious: How did you decide between the separate TrimmableProject.props option and adding %(TrimmableProject) metadata in ProjectReferences.props❔

I followed the pattern for `SharedFxDepList` - assumed separate files were the way to go. Should I merge the contents?

> Out of curiosity, do you have an aspiration to eliminate the component's WASM linking test project?

Right now, this test tests trimmability for all projects in the repo include the WebAssembly ones. I didn't do anything to exclude it since it appears to just work. If we can get some confirmation in the coming ways it doesn't get in the way of doing things, we can remove the WASM specific one.

**Comment by dougbu** [comment]
> Should I merge the contents?

Nope. Just wondered about reasonzzz; I have no objections to a separate file, especially because the added information is used in only a couple of places.

I'm doing something similar for #40242 and _require_ a separate file because the info I'm adding isn't specific to project reference providers. That didn't apply here.

**Comment by JamesNK** [comment]
@pranavkm This got closed when I merged my PR. Rebase and open a new PR and I'll approve 🙏

**Pull request closed without merging**